testing/upload_validation.ts

```diff
@@ -616,7 +618,7 @@ const imageAttributionNodeSchema = z.object({
 const basePropertyValidators = {
   description: z.string().optional(),
   mutable: z.boolean().optional(),
-  autocomplete: z.boolean().optional(),
+  autocomplete: z.union([z.string(), z.array(z.object({}).passthrough())]).optional(),
```

Comment:
Just confirming that if you declare `autocomplete` on a schema property, the compiled metadata json will contain only a formula metadata object with the `execute` stripped out, and not some accidentally serialization of the `execute` function?

Reply:
`JSON.stringify({'stringProp': 'foo', 'functionProp': () => ['bar']})` returns `'{"stringProp":"foo"}'`, so if someone put an inline autocomplete into a sub-object-schema that we don't transform during `makeSyncTable` it should just get silently dropped during JSON serialization.

The downside is that if there's a bug in `makeSyncTable` we could silently drop actual `autocomplete` functions. Hopefully the unit test will prevent someone from accidentally breaking that

Reply:
Checking my understanding, I think what happens so far is that if you pass a function in the autocomplete field, and it's a top-level property, we'll replace the function with a string that's the autocomplete name. If the value is not a function, we'll leave it alone. If the property is not top-level, we'll leave alone whatever is there.

Then I think zod will error out on some values that were left alone, but allow possibly garbage values through if they have the shape of an array of objects. Is that intentional, the static declarations of return values can be any object because it's valid to return an autocomplete result of any shape? And also if you want to statically declare an array of scalar autocomplete values you have to wrap them in an object to make zod happy?

Maybe we can defer this to another PR, but if we're enforcing a {display, value} kind of shape for static values we should enforce that with zod.

I'm not sure what to do about non-top-level fields, whether to silently allow them through if they have the right shape, or strip them out silently since they're unused, or error out on them. Can handle separately though.

Reply:
> Checking my understanding, I think what happens so far is that if you pass a function in the autocomplete field, and it's a top-level property, we'll replace the function with a string that's the autocomplete name. If the value is not a function, we'll leave it alone. If the property is not top-level, we'll leave alone whatever is there.

Yes, that matches my understanding 

> Then I think zod will error out on some values that were left alone, but allow possibly garbage values through if they have the shape of an array of objects

Yes, was accurate. I've updated the code so that Zod validates the shape correctly for strings, numbers, and booleans, and also allows objects through with any shape.

The updated code has a downside, though: if someone ignores typescript errors and lets Zod catch the problem, the list of errors can be pretty long and potentially hard to make sense of. I think all the union types makes Zod unsure when there's a problem about what the type was supposed to be, and it seems to err on the side of giving more possible errors instead of fewer

My inclination is to go-ahead for now with possibly hard-to-read Zod errors and try to improve them in the future if we find that people in practice are ignoring typescript and the documentation and putting strange nonsense values into the autocomplete field.